cpp/src/gandiva/precompiled/string_ops_test.cc

```diff
@@ -2310,12 +2310,49 @@ TEST(TestStringOps, TestFromHex) {
       ::testing::HasSubstr("Error parsing hex string, one or more bytes are not valid."));
   ctx.Reset();
 }
+
 TEST(TestStringOps, TestSoundex) {
   gandiva::ExecutionContext ctx;
   auto ctx_ptr = reinterpret_cast<int64_t>(&ctx);
   int32_t out_len = 0;
   const char* out;
 
+  out = soundex_utf8(ctx_ptr, "robert", 6, &out_len);
```

Comment:
Add a test case where the input is all numbers
Add a test case where there is only alphabet and the rest are numbers

Reply:
Done